novawalletIntegrationTests/Proxy/ProxySyncIntegrationTests.swift

```diff
@@ -0,0 +1,82 @@
+import XCTest
+@testable import novawallet
+import RobinHood
+
+class ProxySyncIntegrationTests: XCTestCase {
+    func testSync() throws {
+        let storageFacade = SubstrateStorageTestFacade()
+        let userStorageFacade = UserDataStorageTestFacade()
+        let chainRegistry = ChainRegistryFacade.setupForIntegrationTest(with: storageFacade)
+        let substrateAccountId = try? "G4qFCkKu7BiaWFNLXfcdZpY94hndyKnzqY1JtmiSBsTPSxC".toAccountId()
+        let operationQueue = OperationQueue()
+        
+        let managedAccountRepository = AccountRepositoryFactory(storageFacade: userStorageFacade)
+            .createManagedMetaAccountRepository(
+                for: nil,
+                sortDescriptors: [NSSortDescriptor.accountsByOrder]
+            )
+      
+        let wallet = MetaAccountModel(
+            metaId: UUID().uuidString,
+            name: "test",
+            substrateAccountId: substrateAccountId,
+            substrateCryptoType: 0,
+            substratePublicKey: substrateAccountId,
+            ethereumAddress: nil,
+            ethereumPublicKey: nil,
+            chainAccounts: [],
+            type: .watchOnly
+        )
+        
+        let accountItem = ManagedMetaAccountModel(
+            info: wallet,
+            isSelected: true,
+            order: 1
+        )
+
+        let saveWalletOperation = managedAccountRepository
+            .saveOperation({
+                [accountItem]
+            }, { [] })
+        
+        operationQueue.addOperations([saveWalletOperation], waitUntilFinished: true)
+        
+        let syncService = ProxySyncService(
+            chainRegistry: chainRegistry,
+            userDataStorageFacade: userStorageFacade,
+            proxyOperationFactory: ProxyOperationFactory(),
+            metaAccountsRepository: managedAccountRepository,
+            chainsFilter: { $0.chainId == KnowChainId.kusama }
+        )
+        
+        let chainsStore = ChainsStore(chainRegistry: chainRegistry)
+        let allProxyChains = chainsStore.availableChainIds().compactMap {
+            let chain = chainsStore.getChain(for: $0)
+            return chain?.hasProxy == true ? chain : nil
+        }.filter { $0.chainId == KnowChainId.kusama }
```

Comment:
I would make a separate function that runs a test per chain per account. Did we also try on Polkadot?